hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java

```diff
@@ -32,4 +48,73 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
         registry.register(new V0490ScheduleSchema());
         registry.register(new V0570ScheduleSchema());
     }
+
+    @Override
+    public Iterator<ExecutableTxn> iterTxnsForInterval(
+            Instant start,
+            Instant end,
+            Supplier<StoreFactory> cleanupStoreFactory,
+            Function<TransactionBody, Set<Key>> requiredKeysFn,
+            HandleContext context) {
+        var store = cleanupStoreFactory.get().writableStore(WritableScheduleStoreImpl.class);
+        // Filter transactions that are not executed/deleted and have all the required keys
+        var executableTxns = store.getByExpirationBetween(start.getEpochSecond(), end.getEpochSecond()).stream()
+                .filter(schedule -> !schedule.executed() && !schedule.deleted())
+                .filter(schedule -> hasAllRequiredKeys(schedule, requiredKeysFn, context))
+                .toList();
+
+        // Return a custom iterator that supports the remove() method
+        return new Iterator<>() {
+            private int currentIndex = -1; // To keep track of the current element
+            private final List<Schedule> transactions = executableTxns;
+            private ExecutableTxn lastReturned = null;
+
+            @Override
+            public boolean hasNext() {
+                return currentIndex + 1 < transactions.size();
+            }
+
+            @Override
+            public ExecutableTxn next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                lastReturned = toExecutableTxn(transactions.get(++currentIndex));
+                return lastReturned;
+            }
+
+            @Override
+            public void remove() {
+                if (lastReturned == null) {
+                    throw new IllegalStateException("No transaction to remove");
+                }
+
+                // Use the StoreFactory to clean up the transaction
+                store.delete(transactions.get(currentIndex).scheduleId(), Instant.now());
```

Comment:
Each time we are asked to remove the last executed transaction, we should call `cleanupStoreFactory.get()` and use the resulting `storeFactory.writableStore(WritableScheduleStore.class)` to delete it.

Reply:
Changed.